File: app/src/main/java/com/mbmc/fiinfo/ui/adapter/StatAdapter.java
package com.mbmc.fiinfo.ui.adapter;

import android.database.Cursor;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.mbmc.fiinfo.R;
import com.mbmc.fiinfo.data.Event;
import com.mbmc.fiinfo.data.EventInfo;
import com.mbmc.fiinfo.helper.Database;


public class StatAdapter extends BaseCursorAdapter {

    @Override
    public RecyclerView.ViewHolder createCursorViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.view_stat, parent, false);
        return new StatViewHolder(view);
    }

    @Override
    public void bindCursorViewHolder(RecyclerView.ViewHolder holder, int position, Cursor cursor) {
        StatViewHolder viewHolder = (StatViewHolder) holder;

        viewHolder.count.setText(cursor.getString(cursor.getColumnIndexOrThrow("count")));

        Event event = Event.get(cursor.getInt(cursor.getColumnIndexOrThrow(Database.COLUMN_TYPE)));
        EventInfo eventInfo = EventInfo.get(viewHolder.itemView.getContext(), event, cursor);
        viewHolder.type.setImageResource(eventInfo.iconId);
        viewHolder.info.setText(eventInfo.info);
    }


    private static class StatViewHolder extends RecyclerView.ViewHolder{
        TextView count, info;
        ImageView type;

        public StatViewHolder(View itemView) {
            super(itemView);
            count = (TextView) itemView.findViewById(R.id.stat_count);
            type = (ImageView) itemView.findViewById(R.id.stat_type);
            info = (TextView) itemView.findViewById(R.id.stat_info);
        }
    }

}

File: app/src/main/java/com/mbmc/fiinfo/ui/stats/StatAdapter.kt
package com.mbmc.fiinfo.ui.stats

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.mbmc.fiinfo.data.Stat
import com.mbmc.fiinfo.databinding.ItemStatBinding

class StatAdapter : ListAdapter<Stat, StatAdapter.ViewHolder>(Comparator()) {

    inner class ViewHolder(private val binding: ItemStatBinding) :
        RecyclerView.ViewHolder(binding.root) {
        fun bind(stat: Stat?) {
            with(binding) {
                stat?.let {
                    this.stat = stat
                    executePendingBindings()
                }
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
        ViewHolder(ItemStatBinding.inflate(LayoutInflater.from(parent.context), parent, false))

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(getItem(position))
    }
}

private class Comparator : DiffUtil.ItemCallback<Stat>() {
    override fun areItemsTheSame(oldItem: Stat, newItem: Stat): Boolean =
        oldItem == newItem

    override fun areContentsTheSame(oldItem: Stat, newItem: Stat): Boolean =
        oldItem.count == newItem.count
                && oldItem.type == newItem.type
                && oldItem.ssid == newItem.ssid
}